scripts/update-clickup-status.js

```diff
@@ -0,0 +1,111 @@
+const fs = require('fs');
+const axios = require('axios');
+
+const file = fs.readFileSync(process.env.GITHUB_EVENT_PATH, 'utf8');
+const eventPayload = JSON.parse(file);
+
+const statusOrders = [
+  'to do',
+  'doing',
+  'waiting for review',
+  'in functionnal test',
+  'in code review',
+  'to release',
+  'released',
+  'closed',
+];
+
+function getClickUpTaskIdFromTitle(pullRequestTitle) {
+  const index = pullRequestTitle.indexOf('(#') + 2;
+
+  const clickUpTag = pullRequestTitle.substring(index);
+  return clickUpTag.substring(0, clickUpTag.indexOf(')'));
+}
+
+async function updateStatusIfNecessary(tagId, currentStatus, targetStatus) {
+  const currentStatusOrder = statusOrders.indexOf(currentStatus);
+  const targetStatusOrder = statusOrders.indexOf(targetStatus);
+
+  if (currentStatusOrder < targetStatusOrder) {
+    await axios({
+      method: 'PUT',
+      url: `https://api.clickup.com/api/v2/task/${tagId}`,
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: process.env.CLICKUP_API_KEY,
+      },
+      data: {
+        status: targetStatus,
+      },
+    });
+
+    return true;
+  }
+
+  return false;
+}
+
+function isWaitingForReview() {
+  if (eventPayload.pull_request
+    && eventPayload.pull_request.requested_reviewers
+    && eventPayload.pull_request.requested_reviewers.length
+    && eventPayload.pull_request.assignees
+    && eventPayload.pull_request.assignees.length
+  ) {
+    const reviewerLogins = eventPayload.pull_request.requested_reviewers.map(
+      (reviewer) => reviewer.login,
+    );
+    const assigneeLogins = eventPayload.pull_request.assignees.map(
+      (assignee) => assignee.login,
+    );
+
+    return assigneeLogins.some(
+      (assigneeLogin) => reviewerLogins.some(
+        (reviewerLogin) => reviewerLogin === assigneeLogin,
+      ),
+    );
+  }
+
+  return false;
+}
+
+function isInCodeReview() {
+  return eventPayload && !!eventPayload.review;
+}
+
+function isApproved() {
+  if (eventPayload && eventPayload.review) {
+    return eventPayload.review.state === 'APPROVED';
+  }
+  return false;
+}
+
+if (eventPayload && eventPayload.pull_request && eventPayload.pull_request.title) {
+  if (eventPayload.pull_request.title.includes('(#')) {
+    const clickUpTaskId = getClickUpTaskIdFromTitle(eventPayload.pull_request.title);
+
+    let targetStatus = 'doing';
+    if (isWaitingForReview()) {
+      targetStatus = 'waiting for review';
+
+      if (isInCodeReview()) {
+        targetStatus = 'in code review';
+      }
+    }
+
+    if (isApproved()) {
+      targetStatus = 'to release';
+    }
+
+    axios({
```

Comment:
[Suggestion] Shouldn't this be refactored with the previous `PUT` call ?

Reply:
I changed it 😉